src/main/java/com/exactpro/th2/common/event/Event.java

```diff
@@ -266,6 +277,36 @@ public Event messageID(MessageID attachedMessageID) {
         return eventBuilder.build();
     }
 
+    public EventBatch toBatchProto(@Nullable EventID parentID) throws JsonProcessingException {
+        List<com.exactpro.th2.common.grpc.Event> events = toListProto(parentID);
+
+        EventBatch.Builder builder = EventBatch.newBuilder()
+            .addAllEvents(events);
+
+        if (parentID != null && events.size() != 1) {
+            builder.setParentEventId(parentID);
+        }
+
+        return builder.build();
+    }
+
+    public List<EventBatch> toListBatchProto(int maxEventBatchContentSize, @Nullable EventID parentID) throws JsonProcessingException {
```

Comment:
Maybe rename it to `toBatchesProtoWithLimit`?

And also it would be great to have a JavaDoc